examples/lpt_nbody_demo.py

```diff
@@ -12,22 +17,33 @@
 import numpy as np
 from jax.experimental import mesh_utils
 from jax.experimental.shard_map import shard_map
-from jax.sharding import Mesh, NamedSharding
 from jax.sharding import PartitionSpec as P
 from scatter import scatter
 
 import jaxdecomp
 
 
-def _global_to_local_size(mesh_shape, sharding):
+def shmap(f: Callable,
+          in_specs: Specs,
+          out_specs: Specs,
+          check_rep: bool = True,
+          auto: frozenset[AxisName] = frozenset()):
+  """Helper function to create a shard_map function that extracts the mesh from the
+    context."""
+  # Extracts the mesh from the context
+  mesh = mesh_lib.thread_resources.env.physical_mesh
```

Comment:
This seems a bit super hacky
Is it the way the context mesh is used by jax?

Reply:
yeah, it's the way pjit accesses the mesh context. But this is an internal API, so could change without too much notice. It does make things very easy though...